File: ros_basics/src/msg_subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"
#include "ros_basics/greet.h"
#include <iostream>

void callback(const ros_basics::greet::ConstPtr& msg)
{
	ROS_INFO("Received Message: %s", msg->greeting.c_str());
	ROS_INFO("Received Number: %d", msg->number);
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "msg_subscriber");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/greet",10,callback);

	ros::spin();
	return 0;
}

File: ros_basics/scripts/msg_subscriber.py
#!/usr/bin/env python 
import rospy 
from std_msgs.msg import String, Int32
from ros_basics.msg import greet

def callback(data):
	rospy.loginfo("Greetings : %s and Number : %d", data.greeting, data.number)

def listener():
	rospy.init_node("msg_subscriber", anonymous=True)
	rospy.Subscriber('greeting', greet, callback)

	rospy.spin()

if __name__ == '__main__':
	try:
		listener()
	except rospy.ROSInterruptException():
		pass